src/library/scala/collection/TraversableLike.scala

```diff
@@ -142,7 +142,7 @@ trait TraversableLike[+A, +Repr] extends Any
 
   def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Repr, B, That]): That = {
     val b = bf(repr)
-    if (that.isInstanceOf[IndexedSeqLike[_, _]]) b.sizeHint(this, that.seq.size)
+    if (that.isInstanceOf[IndexedSeqLike[_, _]]) sizeHintTo(b, that.seq.size)
```

Comment:
Oh, Got it. Didn't see the implementation of `mutable.Builder` before it. Just for curiosity - Can't we use double dispatch for that?